### DISCUSSION TITLE
autoselect available localization of installer based on windows languange

### DISCUSSION THREAD

**Discussion by Klaus1189** [Xanashi/Icaros#64] [Feature Requests]
Hi,
I downloaded latest release [3.3.1](https://github.com/Xanashi/Icaros/releases/tag/v3.3.1) and it offers english as default, but I know I translated it to german, that there must be also german.
Is it possible to make it autoselect an available language like german to PCs which are using german language? Same for other available languages.

Thank you for all your work

**Comment by Xanashi** [comment]
Hi Klaus, 
Thanks for the translation!

I'm not actually sure. That dialog is generated by Inno, so Inno should be the one auto-detecting the local system language.
I don't know if it's possible to make the Inno installer do this, but I'll see if I can find something about it in their documentation.

**Reply by Klaus1189** [reply]
Something like that perhaps?
https://github.com/Aleksoid1978/MPC-BE/blob/master/distrib/mpc-be_setup.iss#L142C12-L142C12

**Comment by Xanashi** [comment]
Hi Klaus,

So I checked Inno's documentation, and it does have an auto-detect language feature, which is enabled by default:
https://jrsoftware.org/ishelp/index.php?topic=setup_languagedetectionmethod

I haven't changed this value in my installer, so it should use the default value, which is uilanguage.
uilanguage "is the language used in Windows' own dialogs".

So this should be enabled in the Icaros Installer already.

**Comment by Klaus1189** [comment]
I test it again tomorrow on another pc.
